src/ui/components/Menuable/Menuable.tsx

```diff
@@ -0,0 +1,191 @@
+import classNames from "classnames";
+import { MouseEvent as ReactMouseEvent, ReactNode } from "react";
+
+import { DangerModal } from "../DangerModal";
+import { Icon } from "../Icon";
+
+import "./menuable.scss";
+import { useMeuable, ItemProps } from "./useMeunable";
+
+export interface MenuableProps {
+  items: ItemProps[];
+  children?: ReactNode;
+  className?: string;
+  menuClassName?: string;
+  opened?: boolean;
+  onOpen?: () => void;
+  onClose?: () => void;
+}
+
+export const Menuable = ({
+  opened,
+  className,
+  menuClassName,
+  children,
+  items,
+  onOpen,
+  onClose,
+}: MenuableProps): JSX.Element => {
+  const {
+    menuRef,
+    isShowing,
+    path,
+    menuItems,
+    openDangerModal,
+    onItemClick,
+    handleClose,
+    handleGoBack,
+    handleOpen,
+    handleDangerModalOpen,
+    handleDangerModalClose,
+  } = useMeuable({
+    opened,
+    items,
+    onOpen,
+    onClose,
+  });
+
+  return (
+    <div
+      ref={menuRef}
+      className={classNames("menuable", { "menuable--active": isShowing }, className)}
+      data-testid="menu"
+      onClick={(e) => {
+        e.stopPropagation();
+
+        if (isShowing) {
+          handleClose();
+        } else {
+          handleOpen();
+        }
+      }}
+    >
+      {children}
+
+      {isShowing && (
+        <div className={classNames("rounded-xl border border-gray-700 menuable__menu", menuClassName)}>
+          {!!path.length && (
+            <div
+              className={classNames(
+                "text-sm whitespace-nowrap cursor-pointer",
+                "flex flex-row flex-nowrap items-center",
+                "text-gray-500 hover:text-gray-300 hover:bg-gray-900 menuable__menu__item",
+              )}
+              onClick={handleGoBack}
+            >
+              <Icon fontAwesome="fas fa-caret-left" />
+
+              <span className="ml-2">Go back</span>
+            </div>
+          )}
+
+          {menuItems.map((item, i) => (
+            <div key={i}>
+              {item.isDangerItem ? (
+                <div
+                  key={item.label}
+                  className={classNames(
+                    "text-sm whitespace-nowrap",
+                    "flex flex-row flex-nowrap items-center",
+                    "menuable__menu__item hover:bg-gray-900 ",
+                    { "cursor-pointer": !item.disabled },
+                    item.className,
+                  )}
+                  onClick={(e) => handleDangerModalOpen(e)}
+                >
+                  {openDangerModal ? (
+                    <DangerModal
```

Comment:
I'd move modal rendering from items rendering because it creates the same number of modals as number or items.

Reply:
Well, it has to be in the scope of each `item` because the `accept` prop is different, so have to be specific with each danger item. 

```
              accept={(e: ReactMouseEvent) => onItemClick(e, item, i)}
```

```
{openDangerModal ? (
            <ConfirmDangerModal
                 accept={(e: ReactMouseEvent) => onItemClick(e, item, i)}
                 openModal={openDangerModal}
                 reject={handleDangerModalClose}
            />
          ) : null}
```

Reply:
Yes, but you can have selected item and use it with modal which will be a level higher and only one instance of it.
Plus, it's not good for performance to have arrow functions in render (like accept).